### ISSUE TITLE
The backend depends on an unknown CSV file

### ISSUE THREAD

**Issue by gzoritchak** [soaringmeteo/soaringmeteo#90]
Hello,

First of all, I want to congratulate you on this superb tool.

I am trying to run the code locally, but I am not familiar with the expected format for the locations CSV file. Can you provide an example file to facilitate the project setup?

**Comment by julienrf** [comment]
Hey,

Indeed, the project is not really self-contained. This is because we wanted to cover at least the same locations as in soarGFS. Ultimately, we will not use that CSV file anymore, but in the meantime you can use that:

~~~ csv
Lausanne,65,465,757,1,2,56,188,258,576,752,886
~~~

**Comment by julienrf** [comment]
I’ve renamed the title of this issue, which I will close when the full domain of GFS will be covered in soarV2 (the CSV file will not be needed anymore).

**Comment by gzoritchak** [comment]
Thank you for your response.

I tried with this file but got an exception:

```
08:39:44.408 INFO  org.soaringmeteo.gfs.in.ForecastRun - Found last run at 2023-09-29T00Z
08:39:44.410 INFO  org.soaringmeteo.gfs.Soaringmeteo$ - Removing old GRIB files
08:39:44.419 INFO  o.soaringmeteo.gfs.DownloadAndRead$ - Downloading forecast data
08:39:44.442 ERROR org.soaringmeteo.gfs.Soaringmeteo$ - Failed to run soaringmeteo
java.util.NoSuchElementException: key not found: Area(D,116,177,-45,-21)
        at scala.collection.immutable.BitmapIndexedMapNode.apply(HashMap.scala:635)
        at scala.collection.immutable.HashMap.apply(HashMap.scala:132)
        at org.soaringmeteo.gfs.DownloadAndRead$.$anonfun$apply$7(DownloadAndRead.scala:90)
        at scala.concurrent.Future$.$anonfun$traverse$1(Future.scala:861)
```

**Comment by julienrf** [comment]
Ah, sorry about this other issue… This is due to a legacy compatibility with the old soarGFS.

It should work if you apply the following changes:

~~~ diff
--- a/backend/src/main/scala/org/soaringmeteo/gfs/DownloadAndRead.scala
+++ b/backend/src/main/scala/org/soaringmeteo/gfs/DownloadAndRead.scala
@@ -87,7 +87,7 @@ object DownloadAndRead {
       import scala.concurrent.ExecutionContext.Implicits.global
       Future
         .traverse(AreaAndHour.all) { areaAndHour =>
-          val locations = locationsByArea(areaAndHour.area)
+          val locations = locationsByArea.getOrElse(areaAndHour.area, Seq.empty)
           for {
             gribFile  <- download(areaAndHour)
             forecasts <- read(locations, gribFile, areaAndHour.hourOffset)
~~~

**Comment by julienrf** [comment]
I also suggest to change `forecast_length = 1` in the file `backend/src/main/reference.conf` to avoid downloading too much data.

**Comment by julienrf** [comment]
FYI, I’ve created #92 to make the setup easier (no need for a CSV file anymore).

**Comment by julienrf** [comment]
Closed by #92

**Comment by gzoritchak** [comment]
I'm able to launch it now. 
Thanks a lot!
